src/Bicep.Core/Workspaces/SourceFileGroupingBuilder.cs

```diff
@@ -25,12 +23,7 @@ public class SourceFileGroupingBuilder
         private readonly IReadOnlyWorkspace workspace;
 
         private readonly Dictionary<Uri, ResultWithDiagnostic<ISourceFile>> fileResultByUri;
-        // For each .bicep file key, a dictionary (keyed by 'module' or other syntax statement in the .bicep file) of the resolved URI for that syntax statement
-        private readonly Dictionary<BicepSourceFile, Dictionary<IArtifactReferenceSyntax, Result<Uri, UriResolutionError>>> uriResultByBicepSourceFileByArtifactReferenceSyntax;
-        // For each .bicep file key, a dictionary (keyed by implicit provider artifact references in the scope of the .bicep file) of the resolved URI for the provider types data file
-        private readonly Dictionary<BicepSourceFile, Dictionary<ArtifactReference, Result<Uri, UriResolutionError>>> uriResultByBicepSourceFileByArtifactReference;
-        // For each .bicep file key, the bundle of resource type provider artifact descriptors of both explicit and implicit provider
-        private readonly Dictionary<BicepSourceFile, ProviderDescriptorBundleBuilder> providerDescriptorBundleBuilderBySourceFile;
+        private readonly Dictionary<IArtifactReferenceSyntax, ArtifactResolutionInfo> artifactLookup;
```

Comment:
There is a scenario where a provider is specified entirely in the `bicepconfig.json`, for example:
`bicepconfig.json`
```json
{
    "providers": {
        "foo": "some_valid_address"
    },
    "implicitProviders": [
        "foo"
    ]
}
```
`main.bicep`
```bicep
// this is a resource comming from the implicitly imported 'foo' provider
resource t 'Foo.Namespace/fooObjectType@1.0.0' = {...}
```

In this case there is no `IArtifactReferenceSyntax` object to act as a key for the `ArifactReference`. In the original implementation these objects are stored in the `ProviderDescriptorBundle`. However after the refactor they are no longer accessible. The implicit collection of providers will vary on a per BicepFile basis based on proximity to the closest `bicepconfig.json` in the file-structure.